### PR TITLE
Add Documentation Field for Dev and TPM Portfolios

### PR DESCRIPTION
**Pull request by andrew032011**
### Summary 

In response to the updated expectations for TPM and Dev portfolios, we're adding a new field for "Documentation" for both Devs and TPM views. This way, Devs and TPMs are reminded to include documentation, and they have a way of specifying the documentation they updated, whether or not it's in the PRs they submitted.

Update this in the DB schema and add a UI element.

### Notion/Figma Link 
https://www.notion.so/cornelldti/Add-Blurb-about-Linking-Documentation-for-TPM-Portfolio-bbcc3f76ec5844f981e90a7921083e17?pvs=4

### Test Plan 

Admin view:

Screenshot 2023-09-26 at 12 55 42 AM

Developer view:
Screenshot 2023-09-26 at 1 29 32 AM

Screenshot 2023-09-26 at 12 51 30 AM

TPM view:

Screenshot 2023-09-26 at 1 29 03 AM

### CODE REVIEW

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx [hunk 1]
```diff
@@ -93,6 +94,11 @@ const DevPortfolioForm: React.FC = () => {
         headerMsg: 'Paragraph Submission Empty',
         contentMsg: 'Please write something for the paragraph section of the assignment.'
       });
+    } else if (!documentationText) {
+      Emitters.generalError.emit({
+        headerMsg: 'Documentation Empty',
+        contentMsg: 'Please write something for the documentation section of the assignment.'
```
Comment (michelleli01) [inline frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:100]: Thanks for these awesome changes Andrew (so quickly as well) I was wondering if you could make this more clarifying so that developers know that they can specify if the documentation is located within a submitted PR or a separate PR.

**Review by oscarwang20 — APPROVED** [review]
```
        _
       /(|
      (  :
     __\  \  _____
   (____)  `|
  (____)|   |
   (____).__|
    (___)__.|_____
               LGTM!
```

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx [hunk 2]
```diff
@@ -230,6 +241,22 @@ const DevPortfolioForm: React.FC = () => {
...
+  setDocumentationText
+}: {
+  documentationText: string;
+  setDocumentationText: React.Dispatch<React.SetStateAction<string>>;
+}) => (
+  <div>
+    <label className={styles.bold}>
+      Documentation: <span className={styles.red_color}>*</span>
+    </label>
+    <p>Please provide a link to at least one piece of documentation you added/updated.</p>
```
Comment (michelleli01) [inline frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:255]: Hmm, I'm not sure if it's clearer to either have the developer simply specify and say "documentation located in PR #_" or have them re-paste the link

**Review by michelleli01 — COMMENTED** [review]
Looks great Andrew, thank you so much for this PR it will be incredibly helpful to align with our new portfolio breakdown! I just wanted to leave some comments. Sorry again for not mentioning this to IDOL beforehand 😢

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx (hunk 1 above)
Comment (andrew032011) [reply frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:100]: No problem. Fully agreed on that suggestion. I've changed it to say:

> Please provide a link to at least one piece of documentation you added/updated. If it's
> included in the PRs you added above, you may simply write "Documentation located in PR (insert
> PR number here)". If you made a separate PR updating documentation in the codebase, please
> link that PR here.

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx (hunk 2 above)
Comment (andrew032011) [reply frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx]: I'd vote for the first one. If it's already included they can let us know. But I'll make it clear that it's OK if they didn't include it in the PRs, and that they can just do a separate documentation PR to fulfill the requirement too.

Comment (michelleli01) [reply frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx]: That sounds good!

File: frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx (hunk 1 above)
Comment (michelleli01) [reply frontend/src/components/Forms/DevPortfolioForm/DevPortfolioForm.tsx:100]: Awesome! Thanks Andrew!

**Review by michelleli01 — APPROVED** [review]
